cluster/state/types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package state
+
+import (
+	"encoding/json"
+	"time"
+
+	ssz "github.com/ferranbt/fastssz"
+)
+
+//go:generate genssz
+
+// RawDAG is a list of mutations that constitute the raw cluster state DAG.
+type RawDAG []Mutation
+
+// rootHasher indicates that a type can be hashed with a ssz.HashWalker.
+type rootHasher interface {
+	HashTreeRootWith(hw ssz.HashWalker) error
+}
+
+// MutationData is the interface that all mutation data types must implement.
+type MutationData interface {
+	rootHasher
+	json.Marshaler
+}
+
+// Mutation mutates the cluster state.
+type Mutation struct {
+	// Parent is the hash of the parent mutation.
+	Parent [32]byte `ssz:"Bytes32"`
+	// Type is the type of mutation.
+	Type MutationType `ssz:"ByteList[64]"` // TODO(corver): Make this a numbered enum maybe (instead of a string)?.
```

Comment:
yeah, numbered enum makes sense. What benefit does having string `MutationType` provides?..

Reply:
string makes it human readable in the json, otherwise it will be a number in the json.